plugins/outputs/opensearch/opensearch.go

```diff
@@ -0,0 +1,443 @@
+				fields[k] = value
+				continue
+			}
+			if o.FloatHandling == "drop" {
+				continue
+			}
+
+			if math.IsNaN(v) || math.IsInf(v, 1) {
+				fields[k] = o.FloatReplacement
+			} else {
+				fields[k] = -o.FloatReplacement
+			}
+		}
+
+		m := make(map[string]interface{})
+
+		m["@timestamp"] = metric.Time()
+		m["measurement_name"] = name
+		m["tag"] = metric.Tags()
+		m[name] = fields
+
+		body, err := json.Marshal(m)
+		if err != nil {
+			return fmt.Errorf("failed to marshal body: %w", err)
+		}
+
+		bulkIndxrItem := opensearchutil.BulkIndexerItem{
+			Action: "index",
+			Index: indexName,
+			Body:      strings.NewReader(string(body)),
+			OnSuccess: onSucc,
+			OnFailure: onFail,
+		}
+		if o.ForceDocumentID {
+			bulkIndxrItem.DocumentID = getPointID(metric)
+		}
+
+		if o.UsePipeline != "" {
+			if pipelineName := o.getPipelineName(o.pipelineName, o.pipelineTagKeys, metric.Tags()); pipelineName != "" {
+				if indexers[pipelineName] != nil {
+					indexers[pipelineName].Add(ctx, bulkIndxrItem)
+					continue
+				}
+			}
+		}
+		
+		indexers["default"].Add(ctx, bulkIndxrItem)
+	}
+
+	for _, bulkIndxr := range indexers {
+		if err := bulkIndxr.Close(ctx); err != nil {
+			return fmt.Errorf("error sending bulk request to Opensearch: %s", err)
+		}
+
+		// Report the indexer statistics
+		stats := bulkIndxr.Stats()
+		if stats.NumAdded < uint64(len(metrics)) {
+			return fmt.Errorf("Opensearch indexed [%d] documents with [%d] errors", stats.NumAdded, stats.NumFailed)
+		} else {
+			fmt.Printf("Opensearch successfully indexed [%d] documents\n", stats.NumAdded)
+		}
+	}
+
+	return nil
+}
+
+
+// BulkIndexer supports pipeline at config level so seperate indexer instance for each unique pipeline
+func getTargetIndexers(metrics []telegraf.Metric, osInst *Opensearch) map[string]opensearchutil.BulkIndexer {
+	var indexers = make(map[string]opensearchutil.BulkIndexer)
+	
+	if osInst.UsePipeline != "" {
+		for _, metric := range metrics {
+			if pipelineName := osInst.getPipelineName(osInst.pipelineName, osInst.pipelineTagKeys, metric.Tags()); pipelineName != "" {
+				// BulkIndexer supports pipeline at config level not metric level
+				if _, ok := indexers[osInst.pipelineName]; ok {
+					continue
+				}
+				indexers[pipelineName] = createBulkIndexer(osInst, pipelineName)
+			}
+		}
+	} 
+
+	indexers["default"] = createBulkIndexer(osInst, "")
+	return indexers
+}
+
+func createBulkIndexer(osInst *Opensearch, pipelineName string) opensearchutil.BulkIndexer{
+	var bulkIndexerConfig = opensearchutil.BulkIndexerConfig{
+		Client:     osInst.osClient,
+		NumWorkers: 4,      // The number of worker goroutines (default: number of CPUs)
+		FlushBytes: 5e+6,   // The flush threshold in bytes (default: 5M)
+	}
+	if pipelineName != "" {
+		bulkIndexerConfig.Pipeline = pipelineName
+	}
+
+	bulkIndxr, err := opensearchutil.NewBulkIndexer(bulkIndexerConfig)
+	if err != nil {
+		fmt.Printf("ERROR while intantiating Opensearch NewBulkIndexer: %s", err)
```

Comment:
```suggestion
		o.Log.Errorf("error while intantiating Opensearch NewBulkIndexer: %w", err)
```
Also, this method still returns `bulkIndxr`, can the rest of the process continue with it?

Reply:
updated the implementation to it populated only if no error. Flow can partially work if there is at-least one valid indexer instance.